### PR TITLE
Adding a check for cached version #98

### PR DESCRIPTION
**Pull request by betsygeo**
I am addressing issue https://github.com/Azure/k8s-bake/issues/98.

- Added a check for the edge case where the version requested was not cached, therefore cachetoolPath was undefined
- Added appropriate tests for this.

### CODE REVIEW

File: src/kubectl-util.test.ts [hunk 1]
```diff
@@ -163,4 +163,18 @@ describe('Testing all functions in kubectl-util file.', () => {
       })
       expect(toolCache.find).toHaveBeenCalledWith('kubectl', 'v2.0.0')
    })
+
+   test('getKubectlPath() - installs kubectl if version is provided but not cached', async () => {
+      jest.restoreAllMocks()
```
Comment (davidgamero) [inline src/kubectl-util.test.ts:168]: let's move the restoreAllMocks() to an [afterEach](https://jestjs.io/docs/setup-teardown#repeating-setup)

**Review by Tatsinnit — APPROVED** [review]

File: src/kubectl-util.ts [hunk 2]
```diff
@@ -41,10 +41,12 @@ export async function getKubectlPath() {
...
       if (!!version && version != LATEST) {
          const cachedToolPath = toolCache.find(kubectlToolName, version)
-         kubectlPath = path.join(
-            cachedToolPath,
-            kubectlToolName + getExecutableExtension()
-         )
+         if (cachedToolPath) {
+            kubectlPath = path.join(
+               cachedToolPath,
+               kubectlToolName + getExecutableExtension()
```
[the remark this answers is not in the document]
Comment (betsygeo) [reply src/kubectl-util.ts:47]: Thank you Tats! This surely improves readability, and I will keep it in mind for future uses. However, for consistency throughout the file, I will keep it as it is.

File: src/kubectl-util.test.ts [hunk 3]
```diff
@@ -163,4 +165,16 @@ describe('Testing all functions in kubectl-util file.', () => {
       })
       expect(toolCache.find).toHaveBeenCalledWith('kubectl', 'v2.0.0')
    })
+
+   test('getKubectlPath() - installs kubectl if version is provided but not cached', async () => {
+      jest.spyOn(core, 'getInput').mockReturnValue('v1.15.0')
+      jest.spyOn(toolCache, 'find').mockReturnValue(undefined)
+      jest.spyOn(os, 'type').mockReturnValue('Windows_NT')
+
+      const expectedPath = await kubectlUtil.installKubectl('v1.15.0')
```
Comment (bosesuneha) [inline src/kubectl-util.test.ts:174]: Can you mock kubectlUtil.installKubectl to return a fixed path so that test focuses only on how getKubectlPath() behaves, not on the implementation or side effects of installKubectl.

Comment (betsygeo) [reply src/kubectl-util.test.ts:174]: Hi Suneha, Thanks for the suggestion! I actually remember trying to mock installKubectl initially, but it kept calling the real function—probably because it's defined and used within the same module, so the mock didn't override the internal reference. I'm still digging into it and will try a few approaches to make the function mockable.

**Review by bosesuneha — APPROVED** [review]
lgtm

**Comment by bosesuneha** [comment]
@betsygeo could you sign your commits please?

**Review by davidgamero — APPROVED** [review]
lgtm
